router/http/src/main/scala/io/buoyant/router/http/DiscardProxySideEffects.scala

```diff
@@ -0,0 +1,31 @@
+package io.buoyant.router.http
+
+import com.twitter.finagle.{Service, ServiceFactory, SimpleFilter, Stack}
+import com.twitter.finagle.http.{Request, Response}
+
+object DiscardProxySideEffects {
+
+  /**
+   * Strips host from request URI and drops Proxy-* headers
+   * Those side-effects are added by proxy-aware clients
+   */
+  object filter extends SimpleFilter[Request, Response] {
```

Comment:
i know we've done this elsewhere, but now i have a slight preference for making this a class, e.g.

```scala
class DiscardProxySideEffects extends SimpleFilter... {
}

object DiscardProxySideEffects {
  ...
}
```

So that the filter can easily be extended.